File: Leetcode/roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        length = len(s)
        roman_integers = { 
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
            }
        result = 0

        for i in range(0, length - 1):
            curr_roman = roman_integers[s[i]]
            next_roman = roman_integers[s[i + 1]]
            result =  result + curr_roman if curr_roman >= next_roman else result - curr_roman

        return result + roman_integers[s[length - 1]]

File: LeetCode/RomanToInteger.cs
public class Solution
{
    public int RomanToInt(string s)
    {
        Dictionary<char, int> romanValues = new Dictionary<char, int>
        {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        int result = 0;

        for (int i = 0; i < s.Length - 1; ++i)
        {
            int next = i + 1;
            if (romanValues[s[i]] < romanValues[s[next]])
            {
                result -= romanValues[s[i]];
            }
            else
            {
                result += romanValues[s[i]];
            }
        }
        result += romanValues[s[s.Length - 1]];
        return result;
    }
}
